From 69a648ea321c755c9420714802e5cfc96e6e8d24 Mon Sep 17 00:00:00 2001
From: Ivan Iskandar <iiiiskandar14@gmail.com>
Date: Sun, 18 Sep 2016 21:33:18 +0700
Subject: [PATCH 20/26] Expose QS edit item decoration background color

PS2:
More descriptive color name

@nathanchance edit: updated for 7.1 Nougat

Change-Id: I9867ca26e7d9dacad37d2b70180a98fede0fb0e8
Signed-off-by: sub77 <sub77@ymail.com>
---
 packages/SystemUI/res/values/du_colors.xml                          | 3 +++
 .../SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java | 6 ++++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/res/values/du_colors.xml b/packages/SystemUI/res/values/du_colors.xml
index 2017342..86a6c48 100644
--- a/packages/SystemUI/res/values/du_colors.xml
+++ b/packages/SystemUI/res/values/du_colors.xml
@@ -28,4 +28,7 @@
     <!-- Navigation button ripple color -->
     <color name="navbutton_ripple_color">#FFFFFFFF</color>
 
+    <!-- QS edit page background color -->
+    <color name="qs_edit_item_decoration_bg">@*android:color/secondary_device_default_settings</color>
+
 </resources>
diff --git a/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java b/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java
index 12485c3..031fd74 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java
@@ -476,11 +476,10 @@ public class TileAdapter extends RecyclerView.Adapter<Holder> implements TileSta
         private TileItemDecoration(Context context) {
             TypedArray ta =
                     context.obtainStyledAttributes(new int[]{android.R.attr.colorSecondary});
-            mDrawable = new ColorDrawable(ta.getColor(0, 0));
+            mDrawable = new ColorDrawable();
             ta.recycle();
         }
 
-
         @Override
         public void onDraw(Canvas c, RecyclerView parent, State state) {
             super.onDraw(c, parent, state);
@@ -499,6 +498,9 @@ public class TileAdapter extends RecyclerView.Adapter<Holder> implements TileSta
                         .getLayoutParams();
                 final int top = child.getTop() + params.topMargin +
                         Math.round(ViewCompat.getTranslationY(child));
+                // Set drawable color
+                mDrawable.setColor(mContext.getResources().getColor(
+                        R.color.qs_edit_item_decoration_bg));
                 // Draw full width, in case there aren't tiles all the way across.
                 mDrawable.setBounds(0, top, width, bottom);
                 mDrawable.draw(c);
-- 
2.1.4

